### PR TITLE
Add lib injection support for Ruby 3.3

### PR DESCRIPTION
**Pull request by marcotc**
Support for Ruby 3.3 is present in the gem, the only thing needed for injection is to change the installation script to allow it to run with that version of Ruby.

Because the Ruby API does not change in incompatible ways in between patch versions, we declare our compatibility with the API version "3.3.0", thus supporting all 3.3.x versions.

**How to test the change?**

Unsure? Have a question? Request a review!

### CODE REVIEW

File: lib-injection/host_inject.rb [hunk 1]
```diff
@@ -83,11 +83,14 @@ def in_bundle?
           Bundler::SharedHelpers.in_bundle?
         end
 
+        # CRuby does *not* break API or ABI compatibility in TEENY version bumps.
+        # The ABI always sets the TEENY version to zero to reflect that: {MAJOR}.{MINOR}.0.
+        # https://www.ruby-lang.org/en/news/2013/12/21/ruby-version-policy-changes-with-2-1-0/
         def runtime_supported?
-          major, minor, = RUBY_VERSION.split('.')
+          major, minor, = RUBY_VERSION.split('.') # {MAJOR}_{MINOR}_{TEENY}
           ruby_api_version = "#{major}.#{minor}.0"
```
Comment (p-datadog) [inline lib-injection/host_inject.rb:91]: What is the purpose of `.0` here? If only two components are used by the logic why not just have two components here?

**Review by p-datadog — APPROVED** [review]

**Review by TonyCTHsu — CHANGES_REQUESTED** [review]
What is blocking?

- Need a custom Ruby 3.3 image to build gems in a static way
- We are currently dealing with package size constraint, which is a prerequisite of this PR.

**Comment by marcotc** [comment]
@TonyCTHsu
> Need a custom Ruby 3.3 image to build gems in a static way
Where are these images stored?

**Comment by ivoanjo** [comment]
> Where are these images stored?

Afaik the dockerfiles are those in https://github.com/DataDog/dd-trace-rb/tree/master/.gitlab . I'm not sure how the rest of the pipeline is set up.

It may be worth documenting this somewhere? E.g. so it's clear how we can every year add new Ruby versions for the injection.

**Comment by TonyCTHsu** [comment]
Obsolete by: https://github.com/DataDog/dd-trace-rb/pull/4137

**Pull request closed without merging**